pkg/reconciler/serverlessservice/serverlessservice_test.go

```diff
@@ -0,0 +1,212 @@
+		Objects: []runtime.Object{
+			sks("test"),
+			vs("test"),
+		},
+		WantCreates: []runtime.Object{
+			dr("test"),
+		},
+		WantEvents: []string{
+			Eventf(corev1.EventTypeNormal, "Created", "Created DestinationRule %q", "test-private"),
+		},
+	}, {
+		Name: "fix both",
+		Key:  "testing/test",
+		Objects: []runtime.Object{
+			sks("test"),
+			func() *istiov1alpha3.VirtualService {
+				virtualService := vs("test")
+				virtualService.Spec.Hosts = []string{"foo"}
+				return virtualService
+			}(),
+			func() *istiov1alpha3.DestinationRule {
+				destinationRule := dr("test")
+				destinationRule.Spec.Host = "foo"
+				return destinationRule
+			}(),
+		},
+		WantUpdates: []clientgotesting.UpdateActionImpl{{
+			Object: vs("test"),
+		}, {
+			Object: dr("test"),
+		}},
+		WantEvents: []string{
+			Eventf(corev1.EventTypeNormal, "Updated", "Updated VirtualService %s", "testing/test-private"),
+			Eventf(corev1.EventTypeNormal, "Updated", "Updated DestinationRule %s", "testing/test-private"),
+		},
+	}, {
+		Name:    "failure for VirtualService",
+		Key:     "testing/test",
+		WantErr: true,
+		WithReactors: []clientgotesting.ReactionFunc{
+			InduceFailure("create", "virtualservices"),
+		},
+		Objects: []runtime.Object{
+			sks("test"),
+			dr("test"),
+		},
+		WantCreates: []runtime.Object{
+			vs("test"),
+		},
+		WantEvents: []string{
+			Eventf(corev1.EventTypeWarning, "CreationFailed", "Failed to create VirtualService %s: inducing failure for create virtualservices", "testing/test-private"),
+			Eventf(corev1.EventTypeWarning, "InternalError", "failed to reconcile VirtualService: failed to create VirtualService: inducing failure for create virtualservices"),
+		},
+	}, {
+		Name:    "failure for DestinationRule",
+		Key:     "testing/test",
+		WantErr: true,
+		WithReactors: []clientgotesting.ReactionFunc{
+			InduceFailure("create", "destinationrules"),
+		},
+		Objects: []runtime.Object{
+			sks("test"),
+			vs("test"),
+		},
+		WantCreates: []runtime.Object{
+			dr("test"),
+		},
+		WantEvents: []string{
+			Eventf(corev1.EventTypeWarning, "CreationFailed", "Failed to create DestinationRule %s: inducing failure for create destinationrules", "testing/test-private"),
+			Eventf(corev1.EventTypeWarning, "InternalError", "failed to reconcile DestinationRule: failed to create DestinationRule: inducing failure for create destinationrules"),
+		},
+	}}
+	table.Test(t, MakeFactory(func(ctx context.Context, listers *Listers, cmw configmap.Watcher) controller.Reconciler {
+		r := &reconciler{
+			istioclient:           istioclient.Get(ctx),
+			virtualServiceLister:  listers.GetVirtualServiceLister(),
+			destinationRuleLister: listers.GetDestinationRuleLister(),
+		}
+
+		return sksreconciler.NewReconciler(ctx, logging.FromContext(ctx), fakenetworkingclient.Get(ctx),
+			listers.GetServerlessServiceLister(), controller.GetEventRecorder(ctx), r, controller.Options{
+				ConfigStore: &testConfigStore{
+					// Enable reconciling gateway.
```

Comment:
Probably missing something obvious, but not sure what this comment refers to. We reconcile SKS to VS and DR, so no gateway really involved?

Reply:
copy aaaaand pasta :D